src/badfish/main.py

```diff
@@ -1353,115 +1223,174 @@ async def toggle_boot_device(self, device):
+                }
+                if vm_path["count"] > 0:
+                    for m in data["Members"]:
+                        vm_path["members"].append(m.get("@odata.id"))
+            except (ValueError, KeyError):
+                raise BadfishException("Not able to access virtual media config.")
         else:
-            raise BadfishException("No VirtualMedia endpoint found")
-
-        return vms
+            try:
+                vm_path = []
+                for m in data["Members"]:
+                    vm_path.append(m.get("@odata.id"))
+            except (ValueError, KeyError):
+                raise BadfishException("Not able to access virtual media config.")
+        return vm_path
 
     async def check_virtual_media(self):
-        vms = await self.get_virtual_media()
-        for vm in vms:
-            disc_url = "%s%s" % (self.host_uri, vm)
-            disc_response = await self.get_request(disc_url)
+        vm_config = await self.get_virtual_media_config()
+        if self.vendor == "Supermicro":
+            if vm_config.get("count") == 0:
+                self.logger.info("No virtual media mounted.")
+                return False
+            else:
+                vm_config = vm_config["members"]
+
+        inserted = False
+        for vm in vm_config:
+            _uri = "%s%s" % (self.host_uri, vm)
+            _response = await self.get_request(_uri)
             try:
-                raw = await disc_response.text("utf-8", "ignore")
-                disc_data = json.loads(raw.strip())
-                self.logger.info(f"{disc_data.get('Id')}:")
-                self.logger.info(f"    Name: {disc_data.get('Name')}")
-                self.logger.info(f"    ImageName: {disc_data.get('ImageName')}")
-                self.logger.info(f"    Inserted: {disc_data.get('Inserted')}")
+                raw = await _response.text("utf-8", "ignore")
+                _data = json.loads(raw.strip())
+                self.logger.info(f"{_data.get('Id')}:")
+                self.logger.info(f"    Name: {_data.get('Name')}")
+                self.logger.info(f"    ImageName: {_data.get('ImageName')}")
+                self.logger.info(f"    Inserted: {_data.get('Inserted')}")
+                if str(_data.get("Inserted")).lower() == "true" and "CD" in str(_data.get("Id")):
+                    inserted = True
             except ValueError:
-                raise BadfishException(
-                    "There was something wrong getting values for VirtualMedia"
-                )
-
+                raise BadfishException("There was something wrong getting values for VirtualMedia")
+        return inserted
+
+    async def mount_virtual_media(self, host, path):
+        vm_config = await self.get_virtual_media_config()
+        _headers = {"Content-Type": "application/json"}
+        if self.vendor == "Supermicro":
+            _payload = {"Host": host, "Path": path, "Username": "", "Password": ""}
```

Comment:
do you need to pass empty username and password here?

Reply:
Probably yes, it's mostly to ensure that any info isn't left from previous mounting operations.
And in [this](https://www.supermicro.com/manuals/other/RedfishRefGuide.pdf) (page 43) reference Redfish guide from Supermicro they themselves pass empty values on request for mounting ISO.